MoreLinq.Test/DuplicatesTest.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates();
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of(
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement"
+                );
+
+            var duplicates = testingSequence.Duplicates().ToArray();
+
+            Assert.That(duplicates, Contains.Item("DUPLICATED_STRING"));
+            Assert.That(duplicates.AtMost(1), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Multiple_Duplicates_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of(
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement",
+                "SECOND_DUPLICATED_STRING",
+                "SECOND_DUPLICATED_STRING"
+            );
+
+            var duplicates = testingSequence.Duplicates().ToArray();
+
+            Assert.That(duplicates, Contains.Item("DUPLICATED_STRING"));
+            Assert.That(duplicates, Contains.Item("SECOND_DUPLICATED_STRING"));
+            Assert.That(duplicates.AtMost(2), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_True_Comparer_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates(new DummyStringAlwaysTrueComparer()).ToArray();
+
+            Assert.That(duplicates.AtMost(1), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_False_Comparer_Then_Empty_Sequence_Is_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates(new DummyStringAlwaysFalseComparer());
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Multiple_Duplicates_Sequence_With_Custom_Always_False_Comparer_Then_Empty_Sequence_Is_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("DUPLICATED_STRING", "DUPLICATED_STRING", "DUPLICATED_STRING");
+
+            var duplicates = testingSequence.Duplicates(new DummyStringAlwaysFalseComparer());
```

Comment:
You can use [**Delegating**](https://www.nuget.org/packages/Delegating) here instead and remove `DummyStringAlwaysTrueComparer` and `DummyStringAlwaysFalseComparer` entirely:

```suggestion
#pragma warning disable CA1307 // Specify StringComparison for clarity
            var duplicates = testingSequence.Duplicates(Delegating.Delegate.EqualityComparer((_, _) => false, (string s) => s.GetHashCode()));
#pragma warning restore CA1307 // Specify StringComparison for clarity
```

Reply:
done